pkg/sql/catalog/tabledesc/validate.go

```diff
@@ -675,13 +675,31 @@ func (desc *wrapper) ValidateSelf(vea catalog.ValidationErrorAccumulator) {
 
 	// Validate mutations and exit early if any of these are deeply corrupted.
 	{
+		var mutationIDs util.FastIntSet
 		mutationsHaveErrs := false
 		for _, m := range desc.Mutations {
+			mutationIDs.Add(int(m.MutationID))
 			if err := validateMutation(&m); err != nil {
 				vea.Report(err)
 				mutationsHaveErrs = true
 			}
 		}
```

Comment:
nit: Can we add a comment here that says something like "validate one mutation ID corresponds to exactly one mutation job"?